CMakeLists.txt

```diff
@@ -79,6 +79,10 @@ if(NOT WIN32)
 endif()
 if(CORE_SYSTEM_NAME STREQUAL linux)
   option(ENABLE_EVENTCLIENTS    "Enable event clients support?" OFF)
+  # Linkers
+  option(ENABLE_GOLD	"Enable gnu gold linker?" ON)
+  option(ENABLE_LLD	"Enable llvm lld linker?" OFF)
+  option(ENABLE_MOLD	"Enable mold linker?" OFF)
```

Comment:
for some reason one of those needs to be set explicitly with the cmake call  (e.g. `-DENABLE_GOLD=ON`)...

Reply:
the option does set it. Works for me and jenkis.

Reply:
strange it happens only with that refactoring commit. i‘ll try to dig it out.

Reply:
@wsnipex the jenkins log of this pr is also lacking `-- Linker: GNU gold` - see https://jenkins.kodi.tv/job/LINUX/36988/consoleFull

while other pr's have it (example 21044) https://jenkins.kodi.tv/job/LINUX/37001/consoleFull
```
13:28:31 -- System type: Linux
13:28:31 -- Linker: GNU gold
13:28:31 -- Checking to see if CXX compiler accepts flag -gsplit-dwarf
```

this is what i'm experiencing too, it may have to do with the refactoring changes in `ArchSetup.cmake`

Reply:
I checked again and can now reproduce, will look into it

Reply:
I added a fixup, not exactly pretty though. Would it be cleaner to hide those linker options in e.g. platform/linux/linkers.cmake?

Reply:
thanks, this does work.

> Would it be cleaner to hide those linker options in e.g. platform/linux/linkers.cmake?

🤷‍♂️ not sure, but i'm happy to test anything

Reply:
pushed the alternative version